### PR TITLE
Add app-level watch-mode tests

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Adds our first app-level watch-mode tests so we can start testing against some more involved scenarios that @chancancode has been fixing :tada:

### CODE REVIEW

**Review by mansona — APPROVED** [review]
Awesome work 🎉

File: pnpm-lock.yaml
Comment (NullVoxPopuli) [inline pnpm-lock.yaml:1]: oofta. I cannot win today

File: tests/scenarios/watch-mode-test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      });
+    });
+
+    test(`pnpm ember test`, async function (assert) {
+      await waitFor('Serving on');
+      const content = 'TWO IS A GREAT NUMBER< I LKE IT A LOT< IT IS THE POWER OF ALL  OF ELECTRONICS, MATH, ETC';
+
+      assert.false(await checkScripts(/js$/, content), 'file has not been created yet');
+
+      fs.writeFile(path.join(app.dir, 'app/simple-file.js'), `export const two = "${content}";`);
```
Comment (lifeart) [inline tests/scenarios/watch-mode-test.ts:70]: should we here `await` for file creation?

Comment (mansona) [reply tests/scenarios/watch-mode-test.ts:70]: I guess yes technically 😂 but we have a thing just after that that is explicitly looking for the "app rebuild" notification which only happens after the file is written so it's safe to not await here 👍

**Pull request merged**